domino_data/_feature_store/git.py

```diff
@@ -0,0 +1,72 @@
+"""Containing APIs to interact with feast git repository"""
+
+from git import FetchInfo, PushInfo, Repo
+
+from ..logging import logger
+from .exceptions import GitPullError, GitPushError
+
+_INVALID_PULL_FLAGS = [FetchInfo.ERROR, FetchInfo.REJECTED]
+_INVALID_PUSH_FLAGS = [
+    PushInfo.ERROR,
+    PushInfo.REJECTED,
+    PushInfo.REMOTE_FAILURE,
+    PushInfo.REMOTE_REJECTED,
+]
+
+
+def pull_repo(repo: Repo) -> None:
```

Comment:
don't we need to specify a branch and/or commit?

Reply:
That's a good question.  Since this operation will update the domino feature views too, don'r we want to limit it to the default main branch?

Reply:
After rethinking about this, I think we should not limit this in the job script. We are working like an artifactory, but we don't support multiple artifacts from different branches. There should be permission control here who can push feature views to domino.  I added the branch parameter here, feel this is better than commit. Would prefer to always to pull the latest code to sync instead of syncing on a possible outdated commit.